### DISCUSSION TITLE
Help VMIX / COMPANION / VARIABLE

### DISCUSSION THREAD

**Discussion by davidclaverie** [bitfocus/companion#2403] [Q&A]
Hi all
im new user of COMPANION with vMix
i need to do something please

When an input is select for example input 2   --> change page of the compagnion streamdeck for example page 3

so FROM VMIX to COMPANION

Thanks a lot for your help.
David

**Comment by dnmeid** [comment]
You can do this with variables and triggers.
Create a trigger "on variable change" with the variable `vmix:mix_1_program` (or whatever mix you want to track)
Use one of the actions "internal: Surface: Set by index to page" or "internal: Surface: Set serialnumber to page". Check to set the page by variable and use the variable of your mix in the Expression field. You also can add an offset, e.g. "$(vmix:mix_1_program) + 1". Then Input 2 will result in page 3 and input 5 in page 6 and so on.
